boards/arm/warp7_m4/doc/warp7_m4.rst

```diff
@@ -0,0 +1,341 @@
+configurations are:
+
+.. code-block:: none
+
+   "zephyr,flash"
+   - &ddr_code
+   - &tcml_code
+   - &ocram_code
+   - &ocram_s_code
+   - &ocram_pxp_code
+   - &ocram_epdc_code
+
+   "zephyr,sram"
+   - &ddr_sys
+   - &tcmu_sys
+   - &ocram_sys
+   - &ocram_s_sys
+   - &ocram_pxp_sys
+   - &ocram_epdc_sys
+
+
+Below you will find the instructions to load and run Zephyr on M4 from A7 using
+u-boot.
+
+Connect both micro USB interfaces into the PC. In one USB interface you will
+have 2 USB serial ports, the first one is the A7 console and the second is the
+M4 console for Zephyr with both configured to work at 115200 8N1.
+The other USB interface is used to power the CPU and IO boards and is connected
+to the USB OTG interface of the i.MX7S.
+
+After powering up the platform stop the u-boot execution on the A7 core and
+expose the eMMC as mass storage with the following command in the u-boot
+prompt: ``ums 0 mmc 0``. Copy the compiled zephyr.bin to the first FAT
+partition and remove the mounted device on the PC by issuing a "Ctrl+C" in the
+u-boot prompt.
+Set the u-boot environment variables and run the zephyr.bin from the
+appropriated memory configured in the Zephyr compilation:
+
+.. code-block:: console
+
+   setenv bootm4 'fatload mmc 0:1 $m4addr $m4fw && dcache flush && bootaux $m4addr'
+   # TCML
+   setenv m4tcml 'setenv m4fw zephyr.bin; setenv m4addr 0x007F8000'
+   setenv bootm4tcml 'run m4tcml && run bootm4'
+   run bootm4tcml
+   # TCMU
+   setenv m4tcmu 'setenv m4fw zephyr.bin; setenv m4addr 0x00800000'
+   setenv bootm4tcmu 'run m4tcmu && run bootm4'
+   run bootm4tcmu
+   # OCRAM
+   setenv m4ocram 'setenv m4fw zephyr.bin; setenv m4addr 0x00900000'
+   setenv bootm4ocram 'run m4ocram && run bootm4'
+   run bootm4ocram
+   # OCRAM_S
+   setenv m4ocrams 'setenv m4fw zephyr.bin; setenv m4addr 0x00180000'
+   setenv bootm4ocrams 'run m4ocrams && run bootm4'
+   run bootm4ocrams
+   # DDR
+   setenv m4ddr 'setenv m4fw zephyr.bin; setenv m4addr 0x80000000'
+   setenv bootm4ddr 'run m4ddr && run bootm4'
+   run bootm4ddr
+
+
+Debugging
+=========
+
+Download and install `J-Link Tools`_ and `NXP iMX7D Connect CortexM4.JLinkScript`_.
+
+To run Zephyr Binary using J-Link create the following script in order to
+get the Program Counter and Stack Pointer from zephry.bin.
+
+get-pc-sp.sh:
+.. code-block:: console
+
+   #!/bin/sh
+
+   firmware=$1
+
+   pc=$(od -An -N 8 -t x4 $firmware | awk '{print $2;}')
+   sp=$(od -An -N 8 -t x4 $firmware | awk '{print $1;}')
+
+   echo pc=$pc
+   echo sp=$sp
+
+
+Get the SP and PC from firmware binary: ``./get-pc-sp.sh zephyr.bin``
+.. code-block:: console
+
+   pc=00900f01
+   sp=00905020
+
+Plug in the J-Link into the board and PC and run the J-Link command line tool:
+
+.. code-block:: console
+
+   /usr/bin/JLinkExe -device Cortex-M4 -if JTAG -speed 4000 -autoconnect 1 -jtagconf -1,-1 -jlinkscriptfile iMX7D_Connect_CortexM4.JLinkScript
```

Comment:
use a continuation line to avoid wrapping:

    /usr/bin/JLinkExe -device Cortex-M4 -if JTAG \
       -speed 4000 -autoconnect 1 -jtagconf -1,-1 \
       -jlinkscriptfile iMX7D_Connect_CortexM4.JLinkScript